mlflow/tracing/trace_manager.py

```diff
@@ -154,9 +153,6 @@ def get_span_from_id(self, request_id: str, span_id: str) -> Optional[MLflowSpan
 
         return trace.span_dict.get(span_id) if trace else None
 
-    # NB: Caching as this requires a linear search over all spans in the trace and
```

Comment:
Note: Removing caching as it caused issue when `end_trace()` API is called twice. When a trace is popped out, this method should return None but the cache doesn't reflect it. The number of span in a single trace should be at most a few hundreds in normal use case and this method is only called within `end_trace()`, so this shouldn't be a big concern.